ReplicationAcceptance/ReplicationAcceptance/ReplicationAcceptance.m

```diff
@@ -313,6 +331,154 @@ -(void) testPushErrorsWhenLocalDatabaseIsDeleted
+                                                                  target:self.primaryRemoteDatabaseURL];
+    CDTReplicator *pushReplicator =  [self.replicatorFactory oneWay:push error:nil];
+    
+    
+    CDTRunBlocksForReplicatorDelegate *delegate = [[CDTRunBlocksForReplicatorDelegate alloc] init];
+    
+    delegate.changeProgressBlock = ^(CDTReplicator *replicator) {
+        if (replicator.state == CDTReplicatorStateStarted && replicator.changesProcessed > 0) {
+            [self.factory deleteDatastoreNamed:self.datastore.name error:nil];
+        }
+    };
+    
+    __block NSError *delegateError = nil;
+    delegate.errorBlock = ^(CDTReplicator *rep, NSError *info) {
+        delegateError = info;
+    };
+    
+    pushReplicator.delegate = delegate;
+
+    NSError *error;
+    if (![pushReplicator startWithError:&error]) {
+        STFail(@"CDTReplicator -startWithError: %@", error);
+    }
+    
+    if (![pullReplicator startWithError:&error]) {
+        STFail(@"CDTReplicator -startWithError: %@", error);
+    }
+    
+    while (pushReplicator.threadExecuting || pullReplicator.threadExecuting) {
+        [NSThread sleepForTimeInterval:1.0f];
+    }
+    
+    
+    //check pull replicator
+    STAssertTrue(n_docs != (NSUInteger)pullReplicator.changesTotal, @"changesTotal: %ld, n_docs %ld",
+                 pullReplicator.changesTotal, n_docs);
+    
+    STAssertTrue(n_docs != (NSUInteger)pullReplicator.changesProcessed, @"changesProcessed: %ld, n_docs %ld",
+                 pullReplicator.changesProcessed, n_docs);
+
+    STAssertEquals(pullReplicator.state, CDTReplicatorStateError, @"Found: %@, expected: (%@)",
+                   [CDTReplicator stringForReplicatorState:pullReplicator.state],
+                   [CDTReplicator stringForReplicatorState:CDTReplicatorStateError]);
+    
+    STAssertEquals(delegateError.code, CDTReplicatorErrorLocalDatabaseDeleted,
+                   @"Wrong error code: %ld", delegateError.code);
+
+    
+    //check push replicator
+    STAssertTrue(n_docs != (NSUInteger)pushReplicator.changesTotal, @"changesTotal: %ld, n_docs %ld",
+                 pushReplicator.changesTotal, n_docs);
+    
+    STAssertTrue(n_docs != (NSUInteger)pushReplicator.changesProcessed, @"changesProcessed: %ld, n_docs %ld",
+                 pushReplicator.changesProcessed, n_docs);
+    
+    STAssertEquals(pushReplicator.state, CDTReplicatorStateError, @"Found: %@, expected: (%@)",
+                   [CDTReplicator stringForReplicatorState:pushReplicator.state],
+                   [CDTReplicator stringForReplicatorState:CDTReplicatorStateError]);
+    
+    STAssertEquals(delegateError.code, CDTReplicatorErrorLocalDatabaseDeleted,
+                   @"Wrong error code: %ld", delegateError.code);
+    
+}
+
+
+-(void) testSyncReplicationErrorsWhenLocalDatabaseDeleted_pullDelegateDeletes
+{
+    [self createLocalDocs:5000];
+    [self createRemoteDocs:5000 suffixFrom:5000];
```

Comment:
I think you need some notes here that you're creating 5k docs to try and guarantee that the deleted operation happens when you're still replicating. Could we have fewer docs?